### PR TITLE
221 add example for training with the lightning framework

### LINKED ISSUE #221: Example/Tutorial - use MONAI in a PyTorch Lightning workflow

### PR DESCRIPTION
**Pull request by marksgraham**
Fixes #221  .

### Description
Adds a new notebook that demonstrates how to train the example in 'spleen_segmentation_3d' using [Lightning](https://github.com/PyTorchLightning/pytorch-lightning).

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### CODE REVIEW

**Comment by Nic-Ma** [comment]
Hi @marksgraham ,

Thanks very much for your PR here, is it OK to review or still under development?

**Comment by marksgraham** [comment]
Hi @Nic-Ma, please go ahead and review.

**Comment by marksgraham** [comment]
Hi @Nic-Ma,

Here are the updates:
>     1. Please add a brief description to **/examples/README.md**
Done.
>     2. Change the dataset path to be same as the Spleen example instead of your own path:
>        _data_root = '/home/mark/projects/monai/data/Task09_Spleen/'_
Done.
>     3. Missing "blank space" at:
>        3.1 return {'loss':loss, 'log':tensorboard_logs}
>        3.2 'val_dice':value
>        3.3 mean_val_loss,'val_dice':mean_val_dice
>        3.4 return {'log':tensorboard_logs}
Fixed.
>     4. The output of 'compute_meandice' is **a list of number** if validation batch size > 1, so {'val_dice': value} is not single value. And 'mean_val_dice = val_dice / len(outputs)' is not correct if the input image number of the **last iteration < batch size**.
I have fixed this, using the  same logic you use in the original spleen notebook.
>     5. Please delete the training log in the notebook, it's too long to view the file.
>        And it's better to **print out the best validation metrics** and epoch number after training.
This is actually meant to be a one line progress bar that lightning provides. Unfortunately it doesn't appear as one line when viewed through GitHub. There are also some other issues with (e.g. it doesn't work with jupyterlab) so I have turned it off for now and just print out metrics at each epoch and at the end of training.
>     6. Why your validation dataset is different from the original spleen_segmentation_3d notebook?
>        Seems the last evaluation results don't match.
I think this is because glob doesn't guarantee the same file ordering across different systems. If I rerun the original `spleen_segmentation_3d notebook` on my machine I get the same validation set that appears in this lightning notebook. If we want them to be consistent we could add a sort by filename before we select the validation set on both notebooks?
>     7. 'et = Net()', what does the **et** mean here? Maybe a more clear variable name?
This was meant to read net, fixed.
>     8. Unnecessary blank spaces: 'ModelCheckpoint(  filepath...'
Fixed.

Let me know if you spot anything else that needs changing.

File: examples/README.md [hunk 1]
```diff
@@ -38,6 +38,8 @@ The example shows the flexibility of MONAI modules in a PyTorch-based program:
 - 3D UNet, Dice loss function, Mean Dice metric for 3D segmentation task.
 - Sliding window inference.
 - Deterministic training for reproducibility.
+#### [notebooks/spleen_segmentation_3d](./notebooks/spleen_segmentation_3d_lightning.ipynb)
```
Comment (Nic-Ma) [inline examples/README.md]: I think it should be:
[notebooks/spleen_segmentation_3d_lightning]

**Comment by marksgraham** [comment]
Hi @Nic-Ma,
Made those small changes to the README and notebook. 
Thanks!

**Comment by Nic-Ma** [comment]
Hi @marksgraham ,

Cool! Let's wait for a while then merge it if no more comments.
Thanks.
